Fix TextureCache::CacheImage does not fill texture outparam

a1rwulf opened the pull request:
Backport of: https://github.com/xbmc/xbmc/pull/17277

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

zag2me commented:
Probably the biggest bug that Kodi has had for years. So happy to see this backported! Thanks.

a1rwulf committed:
Fix TextureCache::CacheImage does not fill texture outparam

If the texture out parameter is given, it's expected to be
filled.
However when requesting a texture which is already in the
processlist, the method fails to fill the requested texture.

This resulted in a hard to trigger bug, where the default
texture is rendered instead of the real thumbnail, although
the thumbnail is available.
